vendor/views/vendor.py

```diff
@@ -103,7 +103,12 @@ def session_cart(self, request, offer):
         return session_cart
 
     def post(self, request, *args, **kwargs):
-        offer = Offer.objects.get(site=get_site_from_request(request), slug=self.kwargs["slug"])
+        try:
+            offer = Offer.objects.get(site=get_site_from_request(request), slug=self.kwargs["slug"], available=True)
+        except ObjectDoesNotExist:
+            messages.error(_("Offer those not exist or is unavailable"))
```

Comment:
Should this be "Offer does not exist..."

Reply:
dam, thanks @daviddwmd